### PR TITLE
OrgDetailsPage: Refactor OrgDetailsPage and test

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
This PR is my attempt at beginning to address a few different things:
The first is with how we use redux. At the moment, I think we're a little ad-hoc with how and when we decide to use redux vs component state. My preference would be to only use redux for truly global state, and/or for cases where state would otherwise have to be drilled down deep into a component tree, as otherwise it feels like an unnecessary amount of boilerplate and indirection is introduced without much tangible benefit. I'm open to other perspectives of course, so please let me know what you think.
In any case, I've moved the `organization` state from the redux slice of the same name, as it was only used for the `OrgDetailsPage` component, and we really just care about the current org name, anyway. In its place I've moved the async logic into a `useEffect` hook which cuts down on boilerplate a fair amount.

The other thing I want to start talking about with this PR is testing.
In the past we used enzyme for unit testing components, but its "shallow" approach to testing components didn't seem to really provide much confidence that our components were actually working correctly. This led to us making the change to react-testing-library, which I think was a step in the right direction. Unfortunately component testing is still a bit of a pain point for a few different reasons. For one, often times a large amount of setup and ceremony is necessary to write even the simplest tests. One way of addressing this would be to provide some helper functions to do the general setup required in a way that is easily overridden when necessary. This is done in a kind of ad-hoc way for many tests, so it might be worthwhile to see if that work can be consolidated in some way.

The other issue is with what parts of the components and their behavior we're actually testing. For the most part, I think testing the redux parts of a component isn't the best approach, and should be treated as an implementation detail. Instead, the consequences of certain logic should be validated by checking the DOM with the tools react-testing-library provides, and in the case of network behavior, rather than checking that a particular redux action was or wasn't dispatched, or that the state was changed in some way, a better approach might be to mock the server itself. One of the benefits of this approach besides not caring that redux is used in the implementation is that it more closely resembles how the component would behave for real, allowing us to be more confident that the tests are validating realistic behavior.
I've experimented with `msw` in this PR, and it seems to work well (I think some of the setup can be simplified with some helper functions), but I'm keen to know what others think.

tl;dr: Some experiments with redux and testing

### CODE REVIEW

**Review by hugohaggmark — COMMENTED** [review]
I think it's great that you start tackling these tricky questions. The trickiest question is how to decide when to use Redux or not and I think without a very clear answer to this question there  will always be a lot of ad hoc ways to solve the same challenge.

Unfortunately I don't fully agree with the solution for this particular PR with a mix of both Redux and ad hoc api calls within the component, I think this is exactly the pattern and adds to the confusion of when to use Redux or not. 

But I really like `msw` and can see great potential using that.

**Comment by kaydelaney** [comment]
> I don't fully agree with the solution for this particular PR with a mix of both Redux and ad hoc api calls within the component, I think this is exactly the pattern and adds to the confusion of when to use Redux or not.

Haha, yeah I was thinking about exactly this a little after I made the PR 😅 I'm going to have to rethink what the best approach for this particular case is, but more than anything I just want to start a discussion about how, in general, we want to think about redux and state

**Comment by kaydelaney** [comment]
Closing this for now as I'd like to reconsider some aspects of the PR
